x-pack/filebeat/tests/integration/framework_test.go

```diff
@@ -0,0 +1,135 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build integration
+
+package integration
```

Comment:
Is there a particular reason to put it under the `x-pack` path?

Reply:
Yes, all Elastic-Agent related code is here, which includes the manager who will interact with this "framework". Even it it is outside `x-pack`, it will require `x-pack` code to work/be used.

Reply:
This file doesn't look like it depends on anything that is x-pack licensed, but the test itself definitely does.

Reply:
Consider moving this to an Apache license out of the x-pack directory so it can be reused by the OSS parts of Beats.